src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.NativeAot.cs

```diff
@@ -97,7 +97,12 @@ internal static IntPtr LoadBySearch(Assembly callingAssembly, bool searchAssembl
                 else if ((callingAssembly != null) && searchAssemblyDirectory)
                 {
                     // Try to load the module alongside the assembly where the PInvoke was declared.
-                    // This only makes sense in dynamic scenarios (JIT/interpreter), so leaving this out for now.
+                    // For PInvokes where the DllImportSearchPath.AssemblyDirectory is specified, look next to the application.
+                    ret = LoadLibraryHelper(Path.Combine(AppContext.BaseDirectory, currLibNameVariation), dllImportSearchPathFlags, ref errorTracker);
```

Comment:
Do we need to or-in `loadWithAlteredPathFlags` to match https://github.com/dotnet/runtime/blob/e857642fd03f8a6acdf42af7d9a814e320ec8432/src/coreclr/vm/nativelibrary.cpp#L708 ?

Reply:
Yes, I missed that. Will add back.